be/src/storage/lake/compaction_scheduler.cpp

```diff
@@ -432,6 +459,21 @@ Status CompactionScheduler::abort(int64_t txn_id) {
     return Status::NotFound(fmt::format("no compaction task with txn id {}", txn_id));
 }
 
+void CompactionScheduler::abort_all() {
+    for (int i = 0; i < _task_queues.task_queue_size(); ++i) {
```

Comment:
there might be a concurrency problem here, task_queue_size might change

Reply:
Theoretically, expect the `abort_all` called only during the shutdown procedure, where all the services have already shutdown. Nevertheless, worth to deal the case.

Reply:
Take a second look at the implementation, it is safe even if the task_queue_size changed, because the try_get() interface can handle the `i` out of index if the queue size shrink. As for the expansion of the queue, it should be fine as long as no tasks can be queued after stopped (a different topic to take care).